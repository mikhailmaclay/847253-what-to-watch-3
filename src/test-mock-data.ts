import {User, Movie, Review, TeamMember, Notification} from './types';

export const mockUser: User = {
  id: 1,
  email: `JohnnyMnemonyc@mail.ru`,
  name: `Johnny`,
  avatar: `https://htmlacademy-react-3.appspot.com/wtw/static/avatar/4.jpg`,
  myList: null
};

export const mockReview: Review = {
  id: 1,
  user: {id: mockUser.id, fullName: mockUser.name},
  date: `2020-03-24T08:26:50.139Z`,
  rating: 10,
  text: `It was really awesome!`
};

export const mockDirector: TeamMember = {
  fullName: `Wes Anderson`,
  role: `Director`
};

export const mockActor: TeamMember = {
  fullName: `Adrien Brody`,
  role: `Actor`
};

export const mockMovie: Movie = {
  id: 1,
  name: `Dardjeeling Limited`,
  genre: `Adventure`,
  releaseDate: `2007`,
  description: [
    `A year after their father\`s funeral, three brothers travel across India by train in an attempt to bond with each other.`
  ],
  runTime: 91,
  team: [
    mockDirector,
    mockActor
  ],
  poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Dardjeeling_Limited.jpg`,
  images: [
    `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/dardjeeling_limited.jpg`
  ],
  backgrounds: [
    `#AD9F8B`,
    `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/Dardjeeling_Limited.jpg`
  ],
  rating: 3.6,
  scoresCount: 165106,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  isInMyList: false,
  reviews: null
};

export const mockNotification: Notification = {
  title: `Test`,
  content: `Testing message.`
};
