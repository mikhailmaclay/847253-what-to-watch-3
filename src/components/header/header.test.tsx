// Libraries
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
// Components
import Header from './header';
//
import {mockUser} from '../../test-mock-data';

const mockStore = configureStore([]);

describe(`<Header/>`, () => {
  const store = mockStore({
    user: mockUser,
  });

  it(`Should render correctly`, () => {
    const result = renderer.create(
        <Provider store={store}>
          <Header/>
        </Provider>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
