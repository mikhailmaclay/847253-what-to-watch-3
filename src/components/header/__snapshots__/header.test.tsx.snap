// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Should render correctly 1`] = `
<header>
  <div
    className="logo"
  >
    <a
      className="logo__link "
      href={null}
      onClick={null}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <div
      className="user-block__avatar"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <img
          alt="Johnny"
          height="63"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/avatar/4.jpg"
          width="63"
        />
      </a>
    </div>
  </div>
</header>
`;
